File: app/src/main/java/com/br/rafael/TesteAndroidv2/ui/loginScreen/LoginRouter.java
package com.br.rafael.TesteAndroidv2.ui.loginScreen;

import android.content.Intent;

import com.br.rafael.TesteAndroidv2.Util.Constants;
import com.br.rafael.TesteAndroidv2.data.model.LoginResponse;
import com.br.rafael.TesteAndroidv2.ui.statementScreen.StatementActivity;

import java.lang.ref.WeakReference;

interface LoginRouterInput{
    void statementScreen(LoginResponse response);
}

public class LoginRouter  implements LoginRouterInput{


    public WeakReference<LoginActivity> activity;

    @Override
    public void statementScreen(LoginResponse response) {

        Intent intent = new Intent(activity.get(), StatementActivity.class);
        intent.putExtra(Constants.INSTANCE.getExtraLoginResponse(),response);

        activity.get().startActivity(intent);
        activity.get().finish();

    }
}

File: ProjetoTesteSantanderK2/app/src/main/java/com/br/projetotestesantanter/loginscreen/LoginRouter.kt
package com.br.projetotestesantanter.loginscreen

import android.content.Intent
import com.br.projetotestesantanter.api.model.LoginResponse
import java.lang.ref.WeakReference


internal interface LoginRouterInput{
    fun passDataListAccounts(loginResponse: LoginResponse) : Intent
    fun navegationListAccounts(intent: Intent)
}

class LoginRouter : LoginRouterInput {

    var activity: WeakReference<LoginActivity>? = null


    override fun passDataListAccounts(loginResponse: LoginResponse): Intent {

        return Intent()
    }

    override fun navegationListAccounts(intent: Intent) {


        activity?.get()?.startActivity(intent)

    }

}